baybe/recommenders/pure/nonpredictive/base.py

```diff
@@ -39,9 +40,19 @@ def recommend(  # noqa: D102
                 f"consider any objectives, meaning that the argument is ignored.",
                 UnusedObjectWarning,
             )
+        if (
+            pending_measurements is not None
```

Comment:
Also here as reminder until we've decided how to handle pending measurements

Reply:
I've changed the condition for the warning
but im not sure we want it anymore
consider someone configures `allow_recommending_pending_experiments=True` then they would always get this warning

Reply:
not sure if I can follow. The basic requirement for the warning is still that there are pending points in the first place, so what is the problem exactly?

Reply:
Yes if a user both i) sets `allow_recommending_pending_experiments=True` and then ii) provides `pending_experiments` this warning would always be shown -> since the user has explicitly set the setting, a warning seems weird -> this is why im saying im not sure we even want it

before the behavior wasnt configurable hence this warning always kind of made sense, now not sure anymore

Reply:
Perhaps I'm overlooking something but I'm not sure if the logic needs to be that complicated. Let's recap what we are trying to achieve: if someone provides `pending_experiments` but the recommender cannot make any use of those, we want an `UnusedObjectWarning`. So:
* if no `pending_experiments` are given --> definitely no warning.
* If `pending_experiments` are given: if the algorithm actually process them --> no warning. If it does not (because it can't or is configured not to) --> warning

Are we aligned here?

Reply:
yes but where is the aspect of the search space type

Reply:
well, since we are in a non-predictive recommender context: if we are *not* on a discrete search space, the algorithm would not process/consider them, right? Hence back to that one case I mentioned above -> warning. Or is that wrong reasoning?

Reply:
what about this appoach: the warning says and is kind of understandable
`f"Pending experiments were provided but the selected recommender "
  f"'{self.__class__.__name__}' only utilizes this information for "
  f"purely discrete spaces and "
  f"{fields(self.__class__).allow_recommending_pending_experiments.name}"
  f"=False.",`

So this clearly states the condition for which no warning should be thrown (assuming pending points were provided, to shorten things a bit):
`discrete_space AND flag=False` 

So negating that (because thats where we want to throw the warning) yields
`not discrere_sapce OR flag=True` --> this is what I implemented

Reply:
Looking at this, seems reasonable to me. But honestly, at this point I've gone through the logic too often and I'm no longer sure if I'm not overlooking something. And I'm sure you have thought at least if not more about it. So let's try with that 👍🏼 😄